docs/sdks/server-sdks/java.md

```diff
@@ -20,58 +22,57 @@ In your pom.xml, add the SDK package as a dependency:
-| -------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------ | ------------------------------------------- |
-| `experiment` (string)                        | An Eppo experiment key                                                                                                   | "recommendation-algo-allocation-17"         |
-| `subject` (string)                           | An identifier of the subject or user assigned to the experiment variation                                                | UUID                                        |
-| `variation` (string)                         | The experiment variation the subject was assigned to                                                                     | "control"                                   |
-| `timestamp` (Date)                           | The time when the subject was assigned to the variation                                                                  | 2021-06-22T17:35:12.000Z                    |
-| `subjectAttributes` (Map<String, EppoValue>) | A free-form map of metadata about the subject. These attributes are only logged if passed to the SDK assignment function | `Map.of("device", EppoValue.valueOf("iOS")` |
-| `featureFlag` (string)                       | An Eppo feature flag key                                                                                                 | "recommendation-algo"                       |
-| `allocation` (string)                        | An Eppo allocation key                                                                                                   | "allocation-17"                             |
+Eppo's SDK uses an internal cache to ensure that duplicate assignment events are not logged to the data warehouse. While Eppo's analytic engine will automatically deduplicate assignment records, this internal cache prevents firing unnecessary events and can help minimize costs associated with event logging.
 
-:::note
-More details about logging and examples (with Segment, Rudderstack, mParticle, and Snowplow) can be found in the [event logging](/sdks/event-logging/) page.
-:::
 
-## 3. Assign variations
+### Initialize the SDK
+
+Initialize the SDK with a SDK key, which can be generated in the [in the Eppo interface](https://eppo.cloud/feature-flags/keys). Initialization should happen when your application starts up to generate a singleton client instance, once per application lifecycle:
+
+```java
+EppoClientConfig config = EppoClientConfig.builder()
+        .apiKey("<sdk-key>")
+        .assignmentLogger((data) -> System.out.println(data.toString()))
+        .build();
+EppoClient eppoClient = EppoClient.init(config);
```

Comment:
just FYI we removed the builder pattern from `EppoClientConfig` in https://github.com/Eppo-exp/sdk-common-jdk/pull/1 when we moved the shared code to `sdk-common-jdk`
This hasn't been released into `java-server-sdk` yet, but will need to be updated when the next version goes out

Reply:
Sounds good! I'll go ahead and merge these changes since this pr has more to do with the overall structure of the docs and then when those changes are live the Java docs can get updated again.